flink-ml-core/src/main/java/org/apache/flink/ml/linalg/BLAS.java

```diff
@@ -129,6 +129,33 @@ private static double norm2(SparseVector x) {
         return JAVA_BLAS.dnrm2(x.values.length, x.values, 1);
     }
 
+    /** Calculates the p-norm of the vector x. */
+    public static double norm(Vector x, double p) {
+        Preconditions.checkArgument(p >= 1.0, "p value must >= 1.0.");
```

Comment:
How about we update the error message to also print the value of `p`?

e.g., "To compute p-norm of a vector, p should be greater than or equal to 1.0. However the specified p is " + p.